backend/eventloop/application_event_loop/metrics_test.go

```diff
@@ -0,0 +1,147 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("Test for Gitopsdeployment metrics counter", func() {
+
+	Context("Prometheus metrics responds to count of active/failed GitopsDeployments", func() {
+		var err error
+		var workspaceID string
+		var ctx context.Context
+		var scheme *runtime.Scheme
+		var workspace *corev1.Namespace
+		var argocdNamespace *corev1.Namespace
+		var dbQueries db.AllDatabaseQueries
+		var k8sClientOuter client.WithWatch
+		var k8sClient *sharedutil.ProxyClient
+		var kubesystemNamespace *corev1.Namespace
+		var informer sharedutil.ListEventReceiver
+		var gitopsDepl *managedgitopsv1alpha1.GitOpsDeployment
+		var appEventLoopRunnerAction applicationEventLoopRunner_Action
+
+		BeforeEach(func() {
+
+			metrics.ClearMetrics()
+
+			ctx = context.Background()
+			informer = sharedutil.ListEventReceiver{}
+
+			scheme,
+				argocdNamespace,
+				kubesystemNamespace,
+				workspace,
+				err = tests.GenericTestSetup()
+			Expect(err).To(BeNil())
+
+			workspaceID = string(workspace.UID)
+
+			gitopsDepl = &managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "gitops-depl",
+					Namespace: workspace.Name,
+					UID:       uuid.NewUUID(),
+				},
+				Spec: managedgitopsv1alpha1.GitOpsDeploymentSpec{
+					Source: managedgitopsv1alpha1.ApplicationSource{
+						RepoURL:        "https://github.com/abc-org/abc-repo",
+						Path:           "/abc-path",
+						TargetRevision: "abc-commit"},
+					Type: managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated,
+					Destination: managedgitopsv1alpha1.ApplicationDestination{
+						Namespace:   "abc-namespace",
+						Environment: "does-not-exist",
+					},
+				},
+			}
+
+			k8sClientOuter = fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(gitopsDepl, workspace, argocdNamespace, kubesystemNamespace).
+				Build()
+
+			k8sClient = &sharedutil.ProxyClient{
+				InnerClient: k8sClientOuter,
+				Informer:    &informer,
+			}
+
+			dbQueries, err = db.NewUnsafePostgresDBQueries(false, false)
+			Expect(err).To(BeNil())
+
+			appEventLoopRunnerAction = applicationEventLoopRunner_Action{
+				getK8sClientForGitOpsEngineInstance: func(gitopsEngineInstance *db.GitopsEngineInstance) (client.Client, error) {
+					return k8sClient, nil
+				},
+				eventResourceName:           gitopsDepl.Name,
+				eventResourceNamespace:      gitopsDepl.Namespace,
+				workspaceClient:             k8sClient,
+				log:                         log.FromContext(context.Background()),
+				sharedResourceEventLoop:     shared_resource_loop.NewSharedResourceLoop(),
+				workspaceID:                 workspaceID,
+				testOnlySkipCreateOperation: true,
+			}
+		})
```

Comment:
Can you also add a test that tests the non-error case? E.g. creating a valid GitOpsDeployment should increase the total metric, but not increase the error metric.